lib/wanda/messaging/mapper.ex

```diff
@@ -25,30 +29,123 @@ defmodule Wanda.Messaging.Mapper do
     )
   end
 
-  def to_execution_completed(%Result{} = result) do
-    result
-    # TODO: do proper mapping, remove miss
-    |> Miss.Map.from_nested_struct()
-    |> ExecutionCompleted.new!()
+  def to_execution_completed(%Execution.Result{
+        execution_id: execution_id,
+        group_id: group_id,
+        check_results: check_results,
+        result: result
+      }) do
+    ExecutionCompleted.new!(
+      execution_id: execution_id,
+      group_id: group_id,
+      check_results: Enum.map(check_results, &to_check_result/1),
+      result: to_result(result)
+    )
   end
 
   defp to_facts_gathering_requested_target(
-         %Target{agent_id: agent_id, checks: target_checks},
+         %Execution.Target{agent_id: agent_id, checks: target_checks},
          checks
        ) do
     fact_requests =
       checks
       |> Enum.filter(&(&1.id in target_checks))
-      |> Enum.flat_map(fn %Check{id: check_id, facts: facts} ->
+      |> Enum.flat_map(fn %Catalog.Check{id: check_id, facts: facts} ->
         to_fact_requests(check_id, facts)
       end)
 
     FactsGatheringRequestedTarget.new!(agent_id: agent_id, fact_requests: fact_requests)
   end
 
   defp to_fact_requests(check_id, facts) do
-    Enum.map(facts, fn %Fact{name: name, gatherer: gatherer, argument: argument} ->
+    Enum.map(facts, fn %Catalog.Fact{name: name, gatherer: gatherer, argument: argument} ->
       FactRequest.new!(check_id: check_id, name: name, gatherer: gatherer, argument: argument)
     end)
   end
+
+  defp to_check_result(%Execution.CheckResult{
+         check_id: check_id,
+         expectation_results: expectation_results,
+         agents_check_results: agents_check_results,
+         result: result
+       }) do
+    CheckResult.new!(
+      check_id: check_id,
+      agents_check_results: Enum.map(agents_check_results, &to_agent_check_result/1),
+      expectation_results: Enum.map(expectation_results, &to_expectation_result/1),
+      result: to_result(result)
+    )
+  end
+
+  defp to_agent_check_result(%Execution.AgentCheckResult{
+         agent_id: agent_id,
+         facts: facts,
+         expectation_evaluations: expectation_evaluations
+       }) do
+    AgentCheckResult.new!(
+      agent_id: agent_id,
+      facts: Enum.map(facts, &to_fact/1),
+      evaluations: Enum.map(expectation_evaluations, &to_expectation_evaluation/1)
+    )
+  end
+
+  defp to_fact(%Execution.Fact{} = fact) do
+    fact
+    |> Map.from_struct()
+    |> Fact.new!()
```

Comment:
I think that this wouldn't work when the struct is encoded to protobuf. It would throw this error:
```
Got error when encoding Trento.Checks.V1.AgentCheckResult#facts: ** (Protobuf.EncodeError) Trento.Checks.V1.Fact#value should be {key, val},
```

And we would need to map as:
```
Fact.new!(check_id: check_id, name: name, value: {:numeric_value, value})
```
(checking the type of the value, to know if we need a numeric/string/boolean, etc).

This is my experience when I tried to make it work at least

In order to make the test most robust, we would need to try to encode the output of the mapper,